src/main/groovy/de/tracetronic/jenkins/plugins/ecutestexecution/clients/RestApiClientV1.groovy

```diff
@@ -69,32 +70,46 @@ class RestApiClientV1 implements RestApiClient{
      * This method performs the package check for the given test package or project. It creates a check execution order
      * to get the execution ID and execute the package check for this ID.
      * @param testPkgPath the path to the package or project to be checked
+     * @param timeout Time in seconds until the check package execution will be aborted
      * @return CheckPackageResult with the result of the check
-     * @throws ApiException on error status codes.
+     * @throws ApiException on error status codes
+     * @throws TimeoutException on timeout exceeded
      */
-    CheckPackageResult runPackageCheck(String testPkgPath) throws ApiException {
-        CheckReport checkReport
+    CheckPackageResult runPackageCheck(String testPkgPath, int timeout) throws ApiException, TimeoutException {
+        def issues = []
         try {
             ChecksApi apiInstance = new ChecksApi(apiClient)
             CheckExecutionOrder order = new CheckExecutionOrder().filePath(testPkgPath)
             String checkExecutionId = apiInstance.createCheckExecutionOrder(order).getCheckExecutionId()
+
             Closure<Boolean> checkStatus = { CheckExecutionStatus response ->
                 response?.status in [null, 'WAITING', 'RUNNING']
             }
 
-            while (checkStatus(apiInstance.getCheckExecutionStatus(checkExecutionId))) {
+            CheckExecutionStatus checkPackageStatus
+            long endTimeMillis = System.currentTimeMillis() + (long) timeout * 1000L
+            while (checkStatus(checkPackageStatus = apiInstance.getCheckExecutionStatus(checkExecutionId))) {
+                if (timeout > 0 && System.currentTimeMillis() > endTimeMillis) {
+                    break
+                }
                 sleep(1000)
             }
-            checkReport = apiInstance.getCheckResult(checkExecutionId)
+
+            println("CheckStatus is:   " + checkPackageStatus.status)
+            if (checkPackageStatus.status != 'FINISHED' ) {
+                throw new TimeoutException("Check package was aborted during timeout after ${timeout} seconds")
```

Comment:
We do not abort the package check, we just stop checking the status because of the timeout declared.

The message should be something like `Timeout: runPackageCheck took longer than ${timeout} seconds`